main.nf

```diff
@@ -15,6 +15,22 @@ nextflow.enable.dsl = 2
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+//Either genome needs to be set or the parameters gtf (for rsem/salmon) and organism, library and keytype
+//(for pathway analysis) have to be provided separately
+//-->gtf is necessary for rsem and salmon
+if (params.input_type in ["rsem", "salmon"]) {
+    if (!params.genome && !params.gtf) { exit 1, 'Please provide either genome or gtf file!' }
+    else if (!params.gtf) { params.gtf = WorkflowMain.getGenomeAttribute(params, 'gtf') }
```

Comment:
Is there a way to warn if the `gtf` field does not exist in `igenomes`?

Reply:
Good point! Just added an exit condition if this happens; the user will be told to provide an own gtf file or open an issue on github. Should I do the same for the other new parameters?